index.js

```diff
@@ -46,8 +46,14 @@ module.exports = function(babel) {
           meta: {}
         }
 
-        if (state.file.opts.filename.includes('-components')) {
-          options.meta.moduleName = parseModuleName(state.file.opts.filename);
+        let filename = state.file.opts.filename;
+        let filenameParts = filename.split('/');
+        filenameParts.splice(0, 1);
+
+        if (filename.includes('-components')) {
+          options.meta.moduleName = parseModuleName(filename);
+        } else if (filenameParts[2] === 'components' && filenameParts.length > 5) {
+          options.meta.moduleName = parseModuleName(filename);
```

Comment:
I wonder if it would be better if parseModuleName does the filename checking and returns `null` if the filename is irrelevant. So this could be 

```js
let filename = state.file.opts.filename;
let moduleName = parseModuleName(filename);
if (moduleName) {
  options.meta.moduleName = moduleName;
}
```